tests/basic.rs

```diff
@@ -0,0 +1,249 @@
+        }
+    }
+    println!("finished writing to stream");
+    Ok(())
+}
+
+/// Read and write to the stream
+/// Returns the UTF8 String received from stream
+async fn simple_read_write(stream: TcpStream) -> Result<String, std::io::Error> {
+    let ref stream = Arc::new(Mutex::new(stream));
+    let reader_join_handle = tokio::spawn(read_from(Arc::clone(stream)));
+    let writer_join_handle = tokio::spawn(write_to(Arc::clone(stream)));
+    let msg = match reader_join_handle.await {
+        Ok(r) => match r {
+            Ok(s) => s,
+            Err(e) => return Err(e.into())
+        }
+        Err(e) => return Err(e.into())
+    };
+    match writer_join_handle.await {
+        Ok(_) => {},
+        Err(e) => return Err(e.into())
+    }
+    Ok(msg)
+}
+
+async fn run_client(
+    client_config: Config,
+    server_port: u16,
+    start_rx: tokio::sync::oneshot::Receiver<()>,
+) -> Result<String, std::io::Error>
+{
+    let timeout = Duration::from_secs(1);
+    let sleep = sleep(timeout);
+    tokio::pin!(sleep);
+    tokio::select! {
+        _ = &mut sleep => { panic!("Client timed out waiting for server after {:?}", timeout) }
+        _ = start_rx => {}  // Do nothing
+    }
+    let stream = match dtls::dial("udp".to_string(), "127.0.0.1".to_string(), server_port, client_config).await {
+        Ok(stream) => stream,
+        Err(e) => return Err(e),
+    };
+    simple_read_write(stream).await
+}
+
+async fn run_server(
+    server_config: Config,
+    server_port: u16,
+    ready_tx: tokio::sync::oneshot::Sender<()>,
+) -> Result<String, Error>
+{
+    let listener = match dtls::listen("udp".to_string(), "127.0.0.1".to_string(), server_port, server_config).await {
+        Ok(listener) => listener,
+        Err(e) => panic!(e),
+    };
+    match ready_tx.send(()) {
+        Ok(_) => {},
+        Err(e) => panic!(e),
+    }
+    let (stream, _addr) = match listener.accept().await {
+        Ok((stream, addr)) => (stream, addr),
+        Err(e) => panic!(e),
+    };
+    simple_read_write(stream).await
+}
+
+fn check_comms(config: Config) {
+    println!("Checking client server communication:\n{:?}\n", config);
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    rt.block_on( async move {
+        let mut client_complete = false;
+        let mut server_complete = false;
+        let server_port = random_port().await;
+        let (server_ready_tx, server_ready_rx) = tokio::sync::oneshot::channel();
+        let client = tokio::spawn(run_client(
+            config,
+            server_port,
+            server_ready_rx,
+        ));
+        let server = tokio::spawn(run_server(
+            config,
+            server_port,
+            server_ready_tx,
+        ));
+        let timeout = Duration::from_secs(5);
+        let sleep = sleep(timeout);
+        tokio::pin!(sleep);
+        tokio::pin!(client);
+        tokio::pin!(server);
+        while !(server_complete && client_complete) {
+            tokio::select! {
```

Comment:
There is a [runtime error](https://github.com/webrtc-rs/webrtc/pull/47/checks?check_run_id=1717692777#step:4:56).

```
thread 'e2e_mtu' panicked at 'unexpected task state', ...
```